src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -833,6 +839,39 @@ private static ulong CurlSendCallback(IntPtr buffer, ulong size, ulong nitems, I
                 return Interop.Http.CURL_READFUNC_ABORT;
             }
 
+            private static void SetLowSpeedLimit(EasyRequest easy, TimeSpan timeout)
+            {
+                // If there's no send/receive timeout on the handler, the defaults will
+                // remain disabled, and we don't need to worry about setting or resetting.
+                if (!easy._handler._hasSendReceiveTimeout)
+                {
+                    return;
+                }
+
+                // We need to ensure the timeout is set appropriately, which may include
+                // disabling a timeout currently set if one was.
+
+                long limit = 0, time = 0; // 0 == disable
+
+                if (timeout != Timeout.InfiniteTimeSpan)
+                {
+                    double secs = timeout.TotalSeconds;
+                    if (secs >= 1)
+                    {
+                        limit = 1;
```

Comment:
Based on what I read for SPEED_LIMIT and SPEED_TIME, this seems like as long as it's still trickling out at two bytes per second the timeout doesn't actually apply.  Maybe this needs to be long.MaxValue, or some other absurdly high thing, so that way you spend the whole time below SPEED_LIMIT